drizzlepac/haputils/analyze.py

```diff
@@ -425,13 +426,30 @@ def analyze_data(input_file_list, log_level=logutil.logging.DEBUG, type=""):
         # Determine if the image has one of these conditions.  The routine
         # will exit processing upon the first satisfied condition.
 
+        # Check if all science image arrays in the RAW file are filled with zero values
+        non_zero_data_in_array = False # start assuming data is zeros
+        science_ext_ind_array = count_sci_extensions(input_file, return_ind=True)
+        # make sure science extension exists
+        if len(science_ext_ind_array)>0:
+            for sci_ext_ind in science_ext_ind_array:
+                science_data = getdata(input_file, sci_ext_ind)
+                # change flag if good data in any science extensnion array
+                if not np.all(science_data==0):
+                    non_zero_data_in_array = True
+                    log.warning(
```

Comment:
When this if block is True, there is no need for the warning.  Seems like there might be an else block missing for this inner if block.

Reply:
I just saw that as well and corrected it.